src/compiler/sys.ts

```diff
@@ -340,11 +340,20 @@ namespace ts {
                     }
 
                     function fileChanged(curr: any, prev: any) {

```

Comment:
I would be interested to learn how to do this.  Do I just specify that it has to have an `mtime` property or is there something more?